### PR TITLE
[SYCL][Graph] Document new command-list enqueue path

### PR DESCRIPTION
**Pull request by fabiomestre**
UR PR: https://github.com/oneapi-src/unified-runtime/pull/1975

### CODE REVIEW

**Review by EwanC — APPROVED** [review]

**Review by againull — APPROVED** [review]

**Comment by pbalcer** [comment]
@fabiomestre please update, the UR PR was merged.

**Comment by fabiomestre** [comment]
> @fabiomestre please update, the UR PR was merged.

I have updated the UR commits and repo 👍

**Comment by pbalcer** [comment]
`/home/runner/work/llvm/llvm/repo/sycl/doc/design/CommandGraph.md:355: WARNING: 'myst' cross-reference target not found: 'wait-event-path-implementation-details-' [myst.xref_missing]`
failure looks related

**Comment by fabiomestre** [comment]
> `/home/runner/work/llvm/llvm/repo/sycl/doc/design/CommandGraph.md:355: WARNING: 'myst' cross-reference target not found: 'wait-event-path-implementation-details-' [myst.xref_missing]` failure looks related

I thnk that issue is fixed. There 2 other failures but they seem unrelated to this PR

File: sycl/cmake/modules/UnifiedRuntimeTag.cmake [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
-#     Merge pull request #2258 from aarongreig/aaron/tryUseExtensionSubgroupInfo
-#     Use extension version of clGetKernelSubGroupInfo when necessary.
-set(UNIFIED_RUNTIME_TAG 9937d029c7fdcbf101e89f8515f640c145e059c5)
+# commit 20e501a036e5e16368eb5c8f1e92289efb3ea735
+# Merge: dcf35a10 ca878c58
+# Author: Piotr Balcer <piotr.balcer@intel.com>
+# Date:   Thu Nov 21 10:55:59 2024 +0100
+#   Merge pull request #1975 from Bensuo/fabio/immediate_append_exp
+#   [Command-Buffers] Implement new command-list enqueue path
+set(UNIFIED_RUNTIME_TAG 20e501a036e5e16368eb5c8f1e92289efb3ea735)
```
Comment (EwanC) [inline sycl/cmake/modules/UnifiedRuntimeTag.cmake:7]: https://github.com/intel/llvm/pull/15843 has merged, which bumps UR past this tag - so I think we can remove touching the tag from this PR.

Comment (EwanC) [reply sycl/cmake/modules/UnifiedRuntimeTag.cmake]: I've merged in `sycl` to remove this

**Review by aarongreig — APPROVED** [review]

**Comment by EwanC** [comment]
@intel/llvm-gatekeepers This is good to merge

**Pull request merged**